File: administration/src/bp-modules/user-settings/__mocks__/Region.ts
import { Region } from '../../../generated/graphql'

export const getTestRegion = (options: {
  id?: number
  name?: string
  prefix?: string
  activatedForApplication?: boolean
  activatedForCardConfirmationMail?: boolean
  applicationConfirmationMailNoteActivated?: boolean
  applicationConfirmationMailNote?: string
}): Region => {
  const {
    id,
    name,
    prefix,
    activatedForApplication,
    activatedForCardConfirmationMail,
    applicationConfirmationMailNoteActivated,
    applicationConfirmationMailNote,
  } = options
  return {
    id: id ?? 0,
    name: name ?? 'Augsburg',
    prefix: prefix ?? 'a',
    activatedForApplication: activatedForApplication ?? true,
    activatedForCardConfirmationMail: activatedForCardConfirmationMail ?? true,
    applicationConfirmationMailNoteActivated: applicationConfirmationMailNoteActivated ?? false,
    applicationConfirmationMailNote: applicationConfirmationMailNote ?? null,
  }
}

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/graphql/regions/schema/types/Region.kt
package app.ehrenamtskarte.backend.graphql.regions.schema.types

data class Region(
    val id: Int,
    val prefix: String,
    val name: String,
    val regionIdentifier: String?,
    val dataPrivacyPolicy: String?,
    val activatedForApplication: Boolean,
    val activatedForCardConfirmationMail: Boolean,
    val applicationConfirmationMailNoteActivated: Boolean,
    val applicationConfirmationMailNote: String?,
)
